### ISSUE TITLE
Python tooz tests fail with "Deadline Exceeded" on grpcio>=1.16.0

### ISSUE THREAD

**Issue by cybertron** [grpc/grpc#17478]
### What version of gRPC and what language are you using?
gRPC 1.16.0 and up
Python
  
### What operating system (Linux, Windows, …) and version?
Linux - both Fedora 27 and Ubuntu 16.04
  
### What runtime / compiler are you using (e.g. python version or version of gcc)
Python 2.7, 3.5, 3.6
  
### What did you do?
Run the tooz (https://github.com/openstack/tooz) tests for etcd: tox -e py36-etcd3
 
### What did you expect to see?
 Tests pass
 
### What did you see instead?
A number of failures with tracebacks such as:

    b'Traceback (most recent call last):'
    b'  File "/opt/stack/tooz/.tox/py36-etcd3/lib/python3.6/site-packages/etcd3/client.py", line 46, in handler'
    b'    return f(*args, **kwargs)'
    b'  File "/opt/stack/tooz/.tox/py36-etcd3/lib/python3.6/site-packages/etcd3/client.py", line 679, in lease'
    b'    metadata=self.metadata'
    b'  File "/opt/stack/tooz/.tox/py36-etcd3/lib/python3.6/site-packages/grpc/_channel.py", line 547, in __call__'
    b'    return _end_unary_response_blocking(state, call, False, None)'
    b'  File "/opt/stack/tooz/.tox/py36-etcd3/lib/python3.6/site-packages/grpc/_channel.py", line 466, in _end_unary_response_blocking'
    b'    raise _Rendezvous(state, None, None, deadline)'
    b'grpc._channel._Rendezvous: <_Rendezvous of RPC that terminated with:'
    b'\tstatus = StatusCode.DEADLINE_EXCEEDED'
    b'\tdetails = "Deadline Exceeded"'
    b'\tdebug_error_string = "{"created":"@1544638582.536473019","description":"Failed to create subchannel","file":"src/core/ext/filters/client_channel/client_channel.cc","file_line":2706,"referenced_errors":[{"created":"@1544638582.536468177","description":"Pick Cancelled","file":"src/core/ext/filters/client_channel/lb_policy/pick_first/pick_first.cc","file_line":221,"referenced_errors":[{"created":"@1544638582.536461309","description":"Deadline Exceeded","file":"src/core/ext/filters/deadline/deadline_filter.cc","file_line":69,"grpc_status":4}]}]}"'
    b'>'
    b''
    b'During handling of the above exception, another exception occurred:'
    b''
    b'Traceback (most recent call last):'
    b'  File "/opt/stack/tooz/tooz/tests/__init__.py", line 36, in skip_if_not_implemented'
    b'    return func(*args, **kwargs)'
    b'  File "/opt/stack/tooz/tooz/tests/__init__.py", line 69, in setUp'
    b'    self._coord.start()'
    b'  File "/opt/stack/tooz/tooz/coordination.py", line 690, in start'
    b'    super(CoordinationDriverWithExecutor, self).start(start_heart)'
    b'  File "/opt/stack/tooz/tooz/coordination.py", line 426, in start'
    b'    self._start()'
    b'  File "/opt/stack/tooz/tooz/drivers/etcd3.py", line 143, in _start'
    b'    self._membership_lease = self.client.lease(self.membership_timeout)'
    b'  File "/opt/stack/tooz/.tox/py36-etcd3/lib/python3.6/site-packages/etcd3/client.py", line 48, in handler'
    b'    _translate_exception(exc)'
    b'  File "/opt/stack/tooz/.tox/py36-etcd3/lib/python3.6/site-packages/etcd3/client.py", line 32, in _translate_exception'
    b'    raise exception'
    b'etcd3.exceptions.ConnectionTimeoutError'
    b''

### Anything else we should know about your project / environment?
This does not happen on gRPC 1.15.0 and does not seem to happen when running individual tests. The failures only happen when the full test suite is run at once.

etcd for these tests is started using pifpaf: https://github.com/jd/pifpaf

It is currently pinned to etcd 3.1.3, although I have tried other versions of etcd and they had the same behavior.

**Comment by lidizheng** [comment]
Error is reproducible with Python 2.7. Thank you for reporting!
```bash
python -m virtualenv env
source env/bin/activate
pip install -U tox
git clone git@github.com:openstack/tooz.git
cd tooz
tox -e py27-etcd3
```

**Comment by cybertron** [comment]
I should also note that at some point we will be excluding the problematic versions from tooz's requirements.txt so the reproducer might need to remove the cap from requirements.txt.

https://review.openstack.org/#/c/624508

**Comment by prometheanfire** [comment]
any status updates on this?

**Comment by lidizheng** [comment]
Is the bug still observed in latest version? @prometheanfire 
The system involved in the reproduction case is too much. I don't have the bandwidth currently. But if this problem concerns you. Feel free to help us bisect the problematic gRPC Python commit.

**Comment by cybertron** [comment]
This seems to have been fixed in 1.18.0.
